engine/src/lib.rs

```diff
@@ -304,14 +304,42 @@ mod contract {
     #[no_mangle]
     pub extern "C" fn submit() {
         let io = Runtime;
-        let input = io.read_input().to_vec();
+        let tx_data = io.read_input().to_vec();
+        let current_account_id = io.current_account_id();
+        let state = state::get_state(&io).sdk_unwrap();
+        let relayer_address = predecessor_address(&io.predecessor_account_id());
+        let args = SubmitArgs {
+            tx_data,
+            ..Default::default()
+        };
+        let result = engine::submit(
+            io,
+            &io,
+            &args,
+            state,
+            current_account_id,
+            relayer_address,
+            &mut Runtime,
+        );
+
+        result
+            .map(|res| res.try_to_vec().sdk_expect(errors::ERR_SERIALIZE))
+            .sdk_process();
+    }
+
+    /// Analog of the `submit` function, but waits for the `SubmitArgs` structure rather than
+    /// the array of bytes representing the transaction.
+    #[no_mangle]
+    pub extern "C" fn submit_with_args() {
```

Comment:
Its possible to add a variant to this by deserializing first to the most common used one, then to the not-common one.

Reply:
Are you sure it's a good idea? The `submit` method is invoked pretty frequently. And on every call, firstly, we should try to deserialize bytes into `SubmitArgs`. It's an additional overhead. Is it really worth it?

Reply:
We can't deserialize to the most commonly used one because the commonly used one is just an array of bytes (serialized Ethereum transaction).

Reply:
Its quite negligible honestly.